src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Extensions/ParenthesizedExpressionSyntaxExtensions.cs

```diff
@@ -51,12 +51,22 @@ public static bool CanRemoveParentheses(this ParenthesizedExpressionSyntax node,
 
             // Simplest cases:
             //   ((x)) -> (x)
-            if (expression.IsKind(SyntaxKind.ParenthesizedExpression) ||
+            if (expression.IsKind(SyntaxKind.ParenthesizedExpression) &&
```

Comment:
Not very sure, but I think the check should be && because the comment above indicates that this searches double parentheses. Is that correct or wrong?

Reply:
i believe the || is intentional.  i.e if you call on the outer, or you call on the inner, you will be told that the parens are unnecessary.